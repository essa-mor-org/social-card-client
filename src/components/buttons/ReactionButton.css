.Reaction_button{
	color: var(--Social-color-number);
	text-decoration-line: none;
	font-size: var(--Social-font-meduim);
}

.Reaction_button::before{
	background-image: url(https://www.facebook.com/rsrc.php/v3/y1/r/vZsdP4nVG2m.png);
    background-repeat: no-repeat;
    content: "";
	width: 18px;
	height: 18px;
	margin-right: 6px;
	display: inline-block;
	margin-bottom: -3px;
}

.Reaction_like::before{
	background-size: 20px 1368px;
	background-position: 0 -180px;
}

.Reaction_comment::before{
	background-size: 20px 1368px;
	background-position: 0 -120px;
}

.Reaction_share::before{
	background-size: 20px 1368px;
	background-position: 0 -320px;
}