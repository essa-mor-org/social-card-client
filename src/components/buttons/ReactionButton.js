import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './ReactionButton.css';

function ReactionButton({ className, text, onClick }) {
	const cls = classNames(className, 'Reaction_button');
	return (<a className={cls} href="#" onClick={onClick}>{text}</a>);
}

ReactionButton.propTypes = {
	className: PropTypes.string,
	text: PropTypes.string.isRequired,
	onClick: PropTypes.func
};

export default ReactionButton;