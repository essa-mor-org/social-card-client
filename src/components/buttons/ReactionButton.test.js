import React from 'react';
import ReactionButton from './ReactionButton';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('ReactionButton test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<ReactionButton text="Like" className="like_class"/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - click callback', () => {
		const callBack = jest.fn();
		const wrapper  = shallow(<ReactionButton text="Like" onClick={callBack} />);
		wrapper.simulate('click');
		expect(callBack).toHaveBeenCalledTimes(1);
	  });
});
