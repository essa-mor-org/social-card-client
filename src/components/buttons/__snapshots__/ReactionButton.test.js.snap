// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactionButton test renders correctly 1`] = `
<a
  className="like_class Reaction_button"
  href="javascript:void(0);"
  onClick={undefined}
>
  Like
</a>
`;
